### PR TITLE
Compatibility with SDL v2.0.3

### PR DESCRIPTION
**Pull request by glebm**
RG350 and GCW0 devices have an ancient version of SDL2.

### CODE REVIEW

File: SourceX/storm/storm.cpp [hunk 1]
```diff
@@ -668,13 +671,13 @@ BOOL SVidPlayContinue(void)
...
-#ifdef USE_SDL1
-		sVidAudioQueue->Enqueue(smk_get_audio(SVidSMK, 0), smk_get_audio_size(SVidSMK, 0));
-#else
+#if SDL_VERSION_ATLEAST(2, 0, 4)
 		if (SDL_QueueAudio(deviceId, smk_get_audio(SVidSMK, 0), smk_get_audio_size(SVidSMK, 0)) <= -1) {
 			SDL_Log(SDL_GetError());
 			return false;
 		}
+#else
+	sVidAudioQueue->Enqueue(smk_get_audio(SVidSMK, 0), smk_get_audio_size(SVidSMK, 0));
```
Comment (AJenbo) [inline SourceX/storm/storm.cpp]: Are there any benefits for now always use this?

File: Source/diablo.h [hunk 2]
```diff
@@ -5,6 +5,10 @@
 
 #ifdef USE_SDL1
 #include "sdl2_to_1_2_backports.h"
+#else
+#if SDL_VERSION_ATLEAST(2, 0, 0)
+#include "sdl2_backports.h"
```
Comment (AJenbo) [inline Source/diablo.h]: Any downsides to always loading this for SDL2?

**Comment by AJenbo** [comment]
Not a huge fan of carrying compat layers to a compat library, but this is fairly modest and slots nicely in with the existing SDL1 support. And this will help out number of people that for some reason still stick to Ubuntu 16.04 (SDL2.0.4).

File: Source/diablo.h (hunk 2 above)
Comment (glebm) [reply Source/diablo.h]: No downsides, done

File: SourceX/storm/storm.cpp (hunk 1 above)
Comment (glebm) [reply SourceX/storm/storm.cpp]: It may be better to use the built-in SDL API when it's available but I'm not sure.

File: Source/diablo.h (hunk 2 above)
Comment (AJenbo) [reply Source/diablo.h]: Nice that bring it down to just 11 lines of added complexity from my point of view :)

File: SourceX/storm/storm.cpp (hunk 1 above)
Comment (AJenbo) [reply SourceX/storm/storm.cpp]: Fair enough, I guess we can clean it up at some later date if we learn something more about it.

**Pull request merged**